Graphs/SimpleTextGraph/index.js

```diff
@@ -22,21 +22,26 @@ class SimpleTextGraph extends AbstractGraph {
 
         this.state = {
             fontSize: INITIAL_FONT_SIZE,
+            height: props.height,
+            width: props.width
         }
     }
 
     componentDidMount() {
-        this.checkFontsize()
+        this.checkFontsize();
     }
 
     componentDidUpdate() {
-        this.checkFontsize()
+        this.checkFontsize();
     }
 
-    componentWillReceiveProps(nextProps) {
-        // reset font size on resize
-        if (this.props.height !== nextProps.height || this.props.width !== nextProps.width) {
-            this.setState({ fontSize: INITIAL_FONT_SIZE })
+    static getDerivedStateFromProps(nextProps, prevState) {
+        if (prevState.height !== nextProps.height || prevState.width !== nextProps.width) {
+            return { 
+                fontSize: INITIAL_FONT_SIZE,
+                height: nextProps.height,
+                width: nextProps.width
+            };
         }
```

Comment:
always return null on getDerivedStateFromProps to indicate that no state updates are required.